Error: invalid memory address while using bollmaker.yaml without enabling redis

if I define redis in the bollmarker but forget to enable redis instance, I will see the following error message. 

```
go run ./cmd/bbgo backtest --exchange max --sync-from 2021-08-01 --config config/bollmaker.yaml
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x473e5fd]

goroutine 1 [running]:
github.com/c9s/bbgo/pkg/service.(*RedisPersistenceService).NewStore(0x0, 0xc0002a47b0, 0x22, 0xc0006d1590, 0x3, 0x3, 0x0, 0xc0006d1590)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/service/persistence_redis.go:36 +0x5d
github.com/c9s/bbgo/pkg/bbgo.(*Persistence).Load(0xc0004fc840, 0x4e825a0, 0xc000540fc0, 0xc0006d1590, 0x3, 0x3, 0x100000000000000, 0x6b7d7b0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/bbgo/persistence.go:50 +0xd0
github.com/c9s/bbgo/pkg/strategy/bollmaker.(*Strategy).LoadState(0xc0001b1040, 0x4, 0x50f18ab)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/strategy/bollmaker/strategy.go:227 +0xe8
github.com/c9s/bbgo/pkg/strategy/bollmaker.(*Strategy).Run(0xc0001b1040, 0x527cef0, 0xc0002961c0, 0x527f438, 0xc00028a1c0, 0xc0004f6000, 0x0, 0x0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/strategy/bollmaker/strategy.go:443 +0x458
github.com/c9s/bbgo/pkg/bbgo.(*Trader).RunSingleExchangeStrategy(0xc00010e660, 0x527cef0, 0xc0002961c0, 0x6e49ae8, 0xc0001b1040, 0xc0004f6000, 0x527f438, 0xc00028a1c0, 0xc0006d0240, 0x0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/bbgo/trader.go:256 +0x3d5
github.com/c9s/bbgo/pkg/bbgo.(*Trader).RunAllSingleExchangeStrategy(0xc00010e660, 0x527cef0, 0xc0002961c0, 0x0, 0x0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/bbgo/trader.go:286 +0x23b
github.com/c9s/bbgo/pkg/bbgo.(*Trader).Run(0xc00010e660, 0x527cef0, 0xc0002961c0, 0x0, 0x0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/bbgo/trader.go:307 +0x1a6
github.com/c9s/bbgo/pkg/cmd.glob..func2(0x5a88840, 0xc0002a2310, 0x0, 0x7, 0x0, 0x0)
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/cmd/backtest.go:300 +0x14f8
github.com/spf13/cobra.(*Command).execute(0x5a88840, 0xc0002a22a0, 0x7, 0x7, 0x5a88840, 0xc0002a22a0)
        /Users/kakashi/.go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x5a89560, 0x8, 0x0, 0x0)
        /Users/kakashi/.go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/kakashi/.go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/c9s/bbgo/pkg/cmd.Execute()
        /Users/kakashi/Programs/Github/trading/bbgo/pkg/cmd/root.go:191 +0x499
main.main()
        /Users/kakashi/Programs/Github/trading/bbgo/cmd/bbgo/main.go:8 +0x25
exit status 2
```

--- a/pkg/bbgo/persistence.go
+++ b/pkg/bbgo/persistence.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 
 	"github.com/c9s/bbgo/pkg/service"
+	log "github.com/sirupsen/logrus"
 )
 
 type PersistenceSelector struct {
@@ -27,6 +28,10 @@ func (p *Persistence) backendService(t string) (service.PersistenceService, erro
 		return p.Facade.Json, nil
 
 	case "redis":
+		if p.Facade.Redis == nil {
+			log.Warn("redis persistence is not available, fallback to memory backend")
+			return p.Facade.Memory, nil
+		}
 		return p.Facade.Redis, nil
 
 	case "memory":

